x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/integration/DocumentLevelSecurityTests.java

```diff
@@ -988,6 +1002,125 @@ public void testGlobalAggregation() throws Exception {
+                        not(containsString("green")),
+                        not(containsString("-2")),
+                        not(containsString("-3"))
+                    )
+                );
+                assertHitCount(response, 1);
+                assertSearchHits(response, "1");
+                // fruits
+                StringTerms fruits = response.getAggregations().get("fruits");
+                assertThat(fruits.getBuckets().size(), equalTo(2));
+                List<StringTerms.Bucket> fruitBuckets = fruits.getBuckets();
+                assertTrue(fruitBuckets.stream().anyMatch(bucket -> bucket.getKeyAsString().equals("apple") && bucket.getDocCount() == 1));
+                assertTrue(fruitBuckets.stream().anyMatch(bucket -> bucket.getKeyAsString().equals("grape") && bucket.getDocCount() == 0));
+                // counts
+                LongTerms counts = response.getAggregations().get("counts");
+                assertThat(counts.getBuckets().size(), equalTo(2));
+                List<LongTerms.Bucket> countsBuckets = counts.getBuckets();
+                assertTrue(countsBuckets.stream().anyMatch(bucket -> bucket.getKeyAsString().equals("-1") && bucket.getDocCount() == 1));
+                assertTrue(countsBuckets.stream().anyMatch(bucket -> bucket.getKeyAsString().equals("-4") && bucket.getDocCount() == 0));
+                // colors
+                for (int i = 1; i <= 3; i++) {
+                    StringTerms colors = response.getAggregations().get("colors" + i);
+                    assertThat(colors.getBuckets().size(), equalTo(1));
+                    assertThat(colors.getBuckets().get(0).getKeyAsString(), equalTo("red"));
+                    assertThat(colors.getBuckets().get(0).getDocCount(), equalTo(1L));
+                }
+                // nested
+                StringTerms nested = response.getAggregations().get("nested");
+                assertThat(nested.getBuckets().size(), equalTo(1));
+                assertThat(nested.getBuckets().get(0).getKeyAsString(), equalTo("red"));
+                assertThat(nested.getBuckets().get(0).getDocCount(), equalTo(1L));
+                StringTerms innerFruits = nested.getBuckets().get(0).getAggregations().get("fruits");
+                List<StringTerms.Bucket> innerFruitsBuckets = innerFruits.getBuckets();
+                assertTrue(innerFruitsBuckets.stream().anyMatch(b -> b.getKeyAsString().equals("apple") && b.getDocCount() == 1));
+                assertTrue(innerFruitsBuckets.stream().anyMatch(b -> b.getKeyAsString().equals("grape") && b.getDocCount() == 0));
+                String currentBucketKey = innerFruitsBuckets.get(0).getKeyAsString(); // don't rely on order
+                assertThat(innerFruitsBuckets.size(), equalTo(2));
+                LongTerms innerCounts = innerFruitsBuckets.get(0).getAggregations().get("counts");
+                assertThat(innerCounts.getBuckets().size(), equalTo(1));
```

Comment:
As noted on Slack, this sometimes comes out 2 instead of 1 (https://gradle-enterprise.elastic.co/s/erjv373hqde6g) with a count of `0` for `-4`. I did not dig further into this yet

Reply:
fixed in https://github.com/elastic/elasticsearch/pull/105714/commits/870263fb3a99354c5346bcc7f0ca64bfd1ee0b6a